astropy/units/format/cds.py

```diff
@@ -106,8 +106,10 @@ def t_X(t):  # multiplication for factor in front of unit
             r"[x×]"
             return t
 
+        # Most units are just combinations of letters with no numbers, but there
+        # are a few special ones (\h is Planch constant) and three that end in 0.
         def t_UNIT(t):
-            r"\%|°|\\h|((?!\d)\w)+"
+            r"%|°|\\h|(a|eps|mu)0|((?!\d)\w)+"
```

Comment:
These are just documentation, right?

Reply:
No! This docstring is used by the lexer as a regex to capture things that are to be considered units. Since `!\d` excludes digits, I had to put in the special case for `a0`, etc.

Reply:
Ok! I recall this now from when I last looked through the formatting code in detail. I thought then as I think now that it's surprising to have docstrings with real code effects. It's not unreasonable in this circumstance, but perhaps a good refactor is to reformat this to be an instance of a class with a `__call__` method.

```python

class LexerFunc:
    regex: ...
    func: Callable[[T], R]
    __slots__ = ("regex", "func")
    def __call__(self, t): return self.func(t)

def _t_Unit(t):
    t.value = cls._get_unit(t)
    return t

t_Unit = LexerFunc(r"%|°|\\h|(a|eps|mu)0|((?!\d)\w)+", _t_Unit)

```
(`units/format/` appears to be a good place to apply mypyc and significantly reduce python function overhead)

This is not to be done in this PR, of course!

Reply:
(though we should probably move this to a separate Issue) @eerovaher, you've been upgrading `units.format` recently. Thoughts?

Reply:
A separate issue may be a good idea - at the same time, we should also really update to the latest PLY - #13399

Reply:
The requirement that the regexes must be in docstrings comes from PLY. Although we bundle it, it's still an external package, so we don't have full control over the API here.